src/lxc/commands.c

```diff
@@ -711,29 +711,25 @@ int lxc_cmd_console(const char *name, int *ttynum, int *fd, const char *lxcpath)
 	if (cmd.rsp.ret < 0) {
 		errno = -cmd.rsp.ret;
 		SYSERROR("Denied access to tty");
-		ret = -1;
-		goto out;
+		return -1;
```

Comment:
Previously this case freed `cmd.rsp.data`. The `rspdata` assignment now happens only below though. I do think there are cases where this can make a difference, though I haven't checked all the code paths. At least `lxc_cmd_rsp_recv` allocates it only based on the returned `datalen` independent of `rsp.ret` except for the `LXC_CMD_CONSOLE` case.

Reply:
We can just move the `rspdata` assignment up. I did this now. Will force-push now.